pkg/commands/internal/migrate/migrate_linter_names.go

```diff
@@ -0,0 +1,922 @@
+				results = append(results, linter)
+				break
+			}
+		}
+	}
+
+	return results
+}
+
+func filter(linters []LinterInfo, fns ...fnFilter) []LinterInfo {
+	var results []LinterInfo
+
+	for _, linter := range linters {
+		if mergeFilters(linter, fns) {
+			results = append(results, linter)
+		}
+	}
+
+	return results
+}
+
+func mergeFilters(linter LinterInfo, fns []fnFilter) bool {
+	for _, fn := range fns {
+		if !fn(linter) {
+			return false
+		}
+	}
+
+	return true
+}
+
+type fnFilter func(linter LinterInfo) bool
+
+func onlyPresets(old versionone.Linters) fnFilter {
+	return func(linter LinterInfo) bool {
+		return linter.hasPresets(old.Presets)
+	}
+}
+
+func onlyDefault(linter LinterInfo) bool {
+	return linter.Default
+}
+
+func notDefault(linter LinterInfo) bool {
+	return !linter.Default
+}
+
+func keepFast(old versionone.Linters) fnFilter {
+	return func(linter LinterInfo) bool {
+		if !ptr.Deref(old.Fast) {
+			return true
+		}
+
+		return !linter.Slow
+	}
+}
+
+func keepSlow(old versionone.Linters) fnFilter {
+	return func(linter LinterInfo) bool {
+		if !ptr.Deref(old.Fast) {
+			return false
+		}
+
+		return linter.Slow
+	}
+}
+
+func unknownLinterNames(names []string, linters []LinterInfo) []string {
+	deprecatedLinters := []string{
+		"deadcode",
+		"execinquery",
+		"exhaustivestruct",
+		"exportloopref",
+		"golint",
+		"ifshort",
+		"interfacer",
+		"maligned",
+		"nosnakecase",
+		"scopelint",
+		"structcheck",
+		"tenv",
+		"varcheck",
+	}
+
+	var results []string
+
+	for _, name := range names {
+		found := slices.ContainsFunc(linters, func(l LinterInfo) bool {
+			return l.isName(name)
+		})
+
+		if !found {
+			if slices.Contains(deprecatedLinters, name) {
+				continue
+			}
+
+			results = append(results, name)
+		}
+	}
+
+	return Unique(results)
+}
+
+func convertStaticcheckLinterNames(names []string) []string {
+	var results []string
+
+	for _, name := range names {
+		if slices.Contains([]string{"stylecheck", "gosimple"}, name) {
+			results = append(results, "staticcheck")
+			continue
+		}
+
+		results = append(results, name)
+	}
+
+	return Unique(results)
+}
+
+func onlyLinterNames(names []string) []string {
+	formatters := []string{"gci", "gofmt", "gofumpt", "goimports"}
+
+	var results []string
+
+	for _, name := range names {
+		if !slices.Contains(formatters, name) {
+			results = append(results, name)
+		}
+	}
+
+	return results
+}
+
+func onlyFormatterNames(names []string) []string {
+	formatters := []string{"gci", "gofmt", "gofumpt", "goimports"}
+
+	var results []string
+
+	for _, name := range names {
+		if slices.Contains(formatters, name) {
+			results = append(results, name)
+		}
+	}
+
+	return results
+}
+
+func convertAlternativeNames(names []string) []string {
+	altNames := map[string]string{
+		"gas":       "gosec",
+		"goerr113":  "err113",
+		"gomnd":     "mnd",
+		"logrlint":  "loggercheck",
+		"megacheck": "staticcheck",
+		"vet":       "govet",
+		"vetshadow": "govet",
+	}
```

Comment:
This map duplicates info from the `allLinters` function. Maybe better is to use the function `allLinters` instead of map?

Reply:
this is not a duplicate.